barman/infofile.py

```diff
@@ -877,3 +877,24 @@ def get_child_backup_info(self, child_backup_id):
                     return backup_info
 
         return None
+
+    def is_full_and_eligible_for_incremental(self):
+        """
+        Used to filter out backups that have a parent backup id and are not considered
+        as FULL backups.
+
+        :note: Only consider backups which are eligible for Postgres core incremental
+        backups
+        backup_method = 'postgres', summarize_wal = 'on', parent_backup_id = None
+
+        :return bool: True if it's a full backup or False if not.
+        :raise barman.exceptions.BackupException: If the backup_method != 'postgres'
+        """
+        if self.backup_manager.config.backup_method != "postgres":
```

Comment:
not sure we want to raise an exception. If the backup method is not Postgres, the backup is not eligible and we just return false and the backup is filtered out of the list. An exception here seems too disruptive IMHO.